pkg/api/v1alpha1/cluster_defaults_test.go

```diff
@@ -154,6 +154,71 @@ func TestSetClusterDefaults(t *testing.T) {
 			},
 			wantErr: "",
 		},
+		{
+			name: "worker node group - no count specified",
+			in: &Cluster{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       ClusterKind,
+					APIVersion: SchemeBuilder.GroupVersion.String(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "eksa-unit-test",
+				},
+				Spec: ClusterSpec{
+					KubernetesVersion: Kube119,
+					ClusterNetwork: ClusterNetwork{
```

Comment:
we might not need these fields to validate count defaults

Reply:
The equality check fails if you don't include the ClusterNetwork. I'm not sure if it's a bug, but the value is set to not-null when it is empty.

Reply:
i dont think its a bug, but the wantCluster network needs to be
```
ClusterNetwork: ClusterNetwork{
    CNIConfig: &CNIConfig{
	Cilium: nil,
    },
},
```
if in does not specify clusterNetwork